src/Action_GIST.cpp

```diff
@@ -231,23 +236,28 @@ Action::RetType Action_GIST::Init(ArgList& actionArgs, ActionInit& init, int deb
   // Grid spacing
   gridspacing_ = actionArgs.getKeyDouble("gridspacn", 0.50);
   // Grid center
-  gridcntr_ = Vec3(0.0);
-  if ( actionArgs.hasKey("gridcntr") ) {
-    gridcntr_[0] = actionArgs.getNextDouble(-1);
-    gridcntr_[1] = actionArgs.getNextDouble(-1);
-    gridcntr_[2] = actionArgs.getNextDouble(-1);
-  } else
+  ArgList centerArgs = actionArgs.GetNstringKey("gridcntr", 3);
+  if ( centerArgs.empty() ) {
     mprintf("Warning: No grid center values specified, using default (origin)\n");
+    gridcntr_ = Vec3(0.0);
+  } else {
+    gridcntr_[0] = centerArgs.getNextDouble(-1);
```

Comment:
I like this change. If we want to be extra careful we can make sure the arguments are valid doubles with the `validDouble()` routine from `StringRoutines.h`, e.g.
```C++
if (!validDouble(centerArgs[0])) {
  mprinterr("Error: '%s' is not a valid argument for 'gridcntr'\n");
  return Action::ERR;
}
gridcntr_[0] = convertToDouble(centerArgs[0]);
```
and so on for the other arguments. Can do something similar for the grid dimensions with `validInteger()`.